### PR TITLE
Feature/word2vec

### PR DESCRIPTION
**Pull request by albert097**

### CODE REVIEW

File: embeddings/embedding/static/word2vec.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
+from dataclasses import dataclass, field
+from typing import Any, Literal, Optional
+
+from embeddings.embedding.static.config import GensimFileConfig, StaticModelHubConfig
+from embeddings.embedding.static.embedding import SingleFileEmbedding, StandardStaticWordEmbedding
+
+
+@dataclass
+class KGR10Word2VecConfig(GensimFileConfig):
```
Comment (djaniak) [inline embeddings/embedding/static/word2vec.py:9]: shouldn't this go to the config.py file?

Comment (djaniak) [reply embeddings/embedding/static/word2vec.py:9]: i see the logic now of putting the fasttext/word2vec code together and leaving the general config and embeddings in separate files

File: embeddings/embedding/static/fasttext.py [hunk 2]
```diff
@@ -26,12 +25,6 @@ def __post_init__(self) -> None:
         self.model_name = f"kgr10.plain.{self.method}.dim{self.dimension}.neg10.bin"
```
Comment (djaniak) [inline embeddings/embedding/static/fasttext.py:26]: in word2vec.py file accessibility is checked: **if not self.file_accessible(self.model_name)**, should it be checked here as well or is it not necessary?

File: embeddings/embedding/static/word2vec.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        ngrams = "mwe" if self.mwe else "plain"
+
+        self.model_name = f"{self.method}.v300.m8.{sampling}.{ngrams}.w2v.gensim"
+        self.ensure_model_accessible(self.model_name)
+
+
+class KGR10Word2VecEmbedding(SingleFileEmbedding, StandardStaticWordEmbedding):
+    @staticmethod
+    def from_config(config: StaticModelHubConfig, **kwargs: Any) -> "KGR10Word2VecEmbedding":
+        if not isinstance(config, KGR10Word2VecConfig):
```
Comment (djaniak) [inline embeddings/embedding/static/word2vec.py:36]: maybe put KGR10Word2VecConfig as the config type if then we raise an error if the type is different?

Comment (albert097) [reply embeddings/embedding/static/word2vec.py:36]: i can not do that, it breaks liskov substitution principle https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides

File: embeddings/embedding/static/fasttext.py (hunk 2 above)
Comment (albert097) [reply embeddings/embedding/static/fasttext.py:27]: it will be added in the next PR

**Review by djaniak — APPROVED** [review]

**Review by laugustyniak — APPROVED** [review]

**Pull request merged**